Add documentation for online vs offline hosts (#7706)
fleetdm/fleet: docs/Using-Fleet/FAQ.md

    - [Scheduled queries](#scheduled-queries)
    - [What are my options for storing the osquery logs?](#what-are-my-options-for-storing-the-osquery-logs)
    - [Troubleshooting](#troubleshooting)
  - [Why does the same query come back faster sometimes?](#why-does-the-same-query-come-back-faster-sometimes)
  - [What happens if I have a query on a team policy and I also have it scheduled to run separately?](#what-happens-if-i-have-a-query-on-a-team-policy-and-i-also-have-it-scheduled-to-run-separately)
  - [Why aren’t my live queries being logged?](#why-arent-my-live-queries-being-logged)
  - [Why does my query work locally with osquery but not in Fleet?](#why-does-my-query-work-locally-with-osquery-but-not-in-fleet)
  - [Can I use the Fleet API to fetch results from a scheduled query pack?](#can-i-use-the-fleet-api-to-fetch-results-from-a-scheduled-query-pack)
  - [How do I automatically add hosts to packs when the hosts enroll to Fleet?](#how-do-i-automatically-add-hosts-to-packs-when-the-hosts-enroll-to-fleet)
  - [How do I automatically assign a host to a team when it enrolls with Fleet?](#how-do-i-automatically-assign-a-host-to-a-team-when-it-enrolls-with-fleet)

>   - [Why my host is not updating a policy's response.](#why-my-host-is-not-updating-a-policys-response)

Reviewer: ```suggestion
  - [Why is my host not updating a policy's response?](#why-is-my-host-not-updating-a-policys-response)
```

  - [What should I do if my computer is showing up as an offline host?](#what-should-i-do-if-my-computer-is-showing-up-as-an-offline-host)
  - [How does Fleet deal with IP duplication?](#how-does-fleet-deal-with-ip-duplication)
  - [Can Orbit run alongside osquery?](#can-orbit-run-alongside-osquery)
  - [Can I control how Orbit handles updates?](#can-i-control-how-orbit-handles-updates)
  - [Can I bundle osquery extensions into Orbit?](#can-i-bundle-osquery-extensions-into-orbit)
  - [What happens to osquery logs if my Fleet server or my logging destination is offline?](#what-happens-to-osquery-logs-if-my-fleet-server-or-my-logging-destination-is-offline)
  - [How does Fleet work with osquery extensions?](#how-does-fleet-work-with-osquery-extensions)
  - [Why do I see "Unknown Certificate Error" when adding hosts to my dev server?](#why-do-i-see-unknown-certificate-error-when-adding-hosts-to-my-dev-server)
  - [Can I hide known vulnerabilities that I feel are insignificant?](#can-i-hide-known-vulnerabilities-that-i-feel-are-insignificant)
  - [Can I create reports based on historical data in Fleet?](#can-i-create-reports-based-on-historical-data-in-fleet)

>   - [When do I need fleetctl vs the REST API vs the Fleet UI?](#when-do-i-need-fleetctl-vs-the-rest-api-vs-the-fleet-ui)

Reviewer: ```suggestion
  - [When do I need fleetctl vs. the REST API vs. the Fleet UI?](#when-do-i-need-fleetctl-vs-the-rest-api-vs-the-fleet-ui)
```

# Using Fleet FAQ

> - [Using Fleet FAQ](#using-fleet-faq)

Author: This and the sub-items in e.g. `[Where are my query results?]` are automatically formatted by my markdown editor. (They attempt to generate the table of contents automatically.)

Let me know if I should revert this.

### CentOS
Run `sudo rpm -e fleet-osquery-X.Y.Z.x86_64`

## How does Fleet determines online and offline status?

> - *Online* hosts are devices that are connected to Fleet and will respond to a live query.

Reviewer: ```suggestion
- **Online** hosts are devices that are connected to Fleet and will respond to a live query.
```

> Fleet considers a host as online if it has connected successfully in the last `distributed_interval` (or `config_tls_refresh`, whichever is smaller) seconds.

Reviewer: Is "seconds" supposed to be at the end of this sentence? It seems confusing since it doesn't specify how many seconds.

Author: Yes, `distributed_interval` is a value e.g. `60` and represents seconds.

Let me know if there's a better way to make this clear.

Reviewer: It looks like you explain that it's 60 seconds in the following sentence, so we could just cut the instance of "seconds" here.

Author: I'll use two examples, let me know if it matches what's explained:

`distributed_interval=10`,`config_tls_refresh=30`
A host is considered online if it has connected to Fleet in the last 70 (10+60) seconds.

`distributed_interval=30`,`config_tls_refresh=20`
Then a host is considered online if it has connected to Fleet in the last 80 (20+60) seconds.

(The +60 seconds is a hard-set buffer.)

Reviewer: That certainly provides more detail. The phrase "hard-set buffer" is confusing me a little . I'd use "set buffer" instead. But if "hard-set buffer" is commonly used, feel free to leave it as is.

Author: By hard-set I mean it's a fixed value of 60 seconds (we also use the term "hardcoded", cannot be changed).

Author: > That certainly provides more detail. 

OK, feel free to make edits or propose alternative ways of explaining this.

Author: I've pushed another try. Let me know if that looks good.

> Fleet considers a host as online if it has connected successfully in the last `distributed_interval` (or `config_tls_refresh`, whichever is smaller) seconds.

Reviewer: ```suggestion
Fleet considers a host as online if it has connected successfully in the last `distributed_interval` (or `config_tls_refresh`, whichever is smaller).
```

A buffer of 60 seconds is added to the calculation to avoid unnecessary flapping between online/offline status (in case hosts take a bit longer than expected to connect to Fleet).
The values for `distributed_interval` and `config_tls_refresh` can be found in the **Settings > Organization settings > Agent options** page for global hosts,
and in the **Settings > Teams > TEAM NAME > Agent options** for hosts that belong to a team.

> - *Offline* hosts are devices that are not online. These hosts may be devices that are turned off or not connected to the internet.

Reviewer: ```suggestion
- **Offline** hosts are devices that are not online. These hosts may be devices that are turned off or not connected to the internet.
```

### CentOS
Run `sudo rpm -e fleet-osquery-X.Y.Z.x86_64`

## How does Fleet determines online and offline status?

### Online hosts

> **Online** hosts are devices that are connected to Fleet and will respond to a live query.

Reviewer 2: ```suggestion
**Online** hosts will respond to a live query.
```

I think we can simplify this. This is because some users may be confused by "Online hosts are devices that are connected to Fleet." I anticipate questions like "Aren't offline hosts still connected?"

> Fleet considers a device as online if it has connected successfully in a window of time set by `distributed_interval` (or `config_tls_refresh`, whichever is smaller).

Reviewer 2: ```suggestion
A host is online if it has connected successfully in a window of time set by `distributed_interval` (or `config_tls_refresh`, whichever is smaller).
```

I think we can simplify this. What do you think?

A buffer of 60 seconds is added to the calculation to avoid unnecessary flapping between online/offline status (in case hosts take a bit longer than expected to connect to Fleet).

> The values for `distributed_interval` and `config_tls_refresh` can be found in the **Settings > Organization settings > Agent options** page for global hosts,

Reviewer: ```suggestion
The values for `distributed_interval` and `config_tls_refresh` can be found in the **Settings > Organization settings > Agent options** page for global hosts
```

and in the **Settings > Teams > TEAM NAME > Agent options** page for hosts that belong to a team.

For example:

`distributed_interval=10, config_tls_refresh=30`
A host is considered online if it has connected to Fleet in the last 70 (10+60) seconds.

`distributed_interval=30, config_tls_refresh=20`
A host is considered online if it has connected to Fleet in the last 80 (20+60) seconds.

### Offline hosts

> **Offline** hosts are devices that are not online. These hosts may be devices that are turned off or not connected to the internet.

Reviewer 2: ```suggestion
**Offline** hosts won't respond to a live query. These hosts may be shut down, asleep, or not connected to the internet.
```

Suggested edits based on the following changes coming to the Fleet UI:
![Screen Shot 2022-09-15 at 9 21 05 AM](https://user-images.githubusercontent.com/47070608/190414545-ce23e5c6-0bdc-4910-af24-ac594ab7b21f.png)

> A host could also be marked as offline if there is a connection issue between the osquery agent running in the host and Fleet (see [What should I do if my computer is showing up as an offline host?](#what-should-i-do-if-my-computer-is-showing-up-as-an-offline-host)).

Reviewer 2: ```suggestion
A host could be offline if there is a connection issue between the osquery agent running in the host and Fleet (see [What should I do if my computer is showing up as an offline host?](#what-should-i-do-if-my-computer-is-showing-up-as-an-offline-host)).
```

Proposed simplification.
